prediction/src/tests/test_endpoints.py

```diff
@@ -4,14 +4,35 @@
 
     Provides unit tests for the API endpoints.
 """
-from functools import partial
 import json
+from functools import partial
+from functools import wraps
 
 import pytest
-
 from flask import url_for
-from src.factory import create_app
 from src.algorithms import classify, identify, segment
+from src.factory import create_app
+
+
+def skip_slow_test(func=None):
+    """
+    Skip the wrapped test function unless the environment variable RUN_SLOW_TESTS is set to 1.
+    """
+
+    def wrapper(func):
+        @wraps(func)
+        def wrapped(*args, **kwargs):
+            import os
+            if os.environ.get('RUN_SLOW_TESTS') != '1':
+                return
```

Comment:
This should throw some kind of ``skip`` exception, not silently return.

Reply:
Maybe I got you wrong but when I simply put a `raise Exception(...)` there, the tests fails. If we try to print or log something, it's captured by pytest und only shown if the respective test fails. So what about normal printing but running pytest with a -s flag, which does not capture stdout but would also show warnings and exception messages of other tests that occur when running them?
It would currently look like this:
```
(venv3.5) ➜  concept-to-clinic git:(1_identify_nodules) ✗ sudo bash tests/test_docker.sh
+ docker-compose -f local.yml run prediction pytest -s
======================================== test session starts ========================================
platform linux -- Python 3.6.2, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /app, inifile:
collecting 3 itemsUsing TensorFlow backend.
collected 22 items 

src/tests/test_calculate_volume.py ...
src/tests/test_classify_trained_model_predict.py .2017-09-19 07:33:33.828938: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-09-19 07:33:33.828960: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-09-19 07:33:33.828965: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-09-19 07:33:33.828969: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-09-19 07:33:33.828973: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
.
src/tests/test_crop_dicom.py .
src/tests/test_endpoints.py ..test_identify has been skipped
......
src/tests/test_load_dicom.py Exception reading *.dcm-files:  File is missing 'DICM' marker. Use force=True to force reading
.Exception extracting voxel data:  Dataset does not have attribute 'ImageOrientationPatient'.
...
src/tests/test_preprocess_dicom.py ....

==================================== 22 passed in 27.18 seconds =====================================
+ docker-compose -f local.yml run interface python manage.py test
WARNING: Found orphan containers (concepttoclinic_compile_docs_1) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Starting concepttoclinic_postgres_1 ... 
Starting concepttoclinic_prediction_1 ... done
Postgres is up - continuing...
/usr/local/lib/python3.6/site-packages/environ/environ.py:618: UserWarning: Error reading /app/.env - if you're not configuring your environment separately, check this.
  "environment separately, check this." % env_file)
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
..........
----------------------------------------------------------------------
Ran 10 tests in 0.073s

OK
Destroying test database for alias 'default'...
+ docker-compose -f local.yml run interface python manage.py makemigrations --dry-run --check
WARNING: Found orphan containers (concepttoclinic_compile_docs_1) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Starting concepttoclinic_postgres_1 ... 
Starting concepttoclinic_postgres_1 ... done
Postgres is up - continuing...
/usr/local/lib/python3.6/site-packages/environ/environ.py:618: UserWarning: Error reading /app/.env - if you're not configuring your environment separately, check this.
  "environment separately, check this." % env_file)
No changes detected

```

Reply:
So you can you pytest here to help you

```python
import pytest

def skip_slow_test():
    return not os.environ.get('RUN_SLOW_TESTS')


@pytest.mark.skipif(skip_slow_test, reason='some reason')
def test_slow_test():
    pass
```

Reply:
Thanks @reubano that's way cleaner! When handing over `-rs` to pytest, the skip reason is shown without the output of other tests being displayed. Nice!